src/components/Profile/index.js

```diff
@@ -0,0 +1,118 @@
+import {useState} from 'react';
+
+import {useStore} from '../../hooks/useStore';
+import Button from '../Button';
+import Dialog from '../Dialog';
+import Form from '../Form';
+
+import StyledProfile from './styled';
+
+export default function ProfileComponent() {
+	const users = useStore(state => state.users);
+	const activeUser = useStore(state => state.activeUser);
+	const setActiveUser = useStore(state => state.setActiveUser);
+	const changeUsername = useStore(state => state.changeUsername);
+	const [toggle, setToggle] = useState(true);
+	const [open, setOpen] = useState(false);
+
+	function changeActiveUser() {
+		users.map(user => {
+			if (user !== activeUser) {
+				setActiveUser(user);
+			}
+			return user;
+		});
+	}
+	const setNewUsername = event => {
+		event.preventDefault();
+		changeUsername(activeUser.userID, event.target.inputField.value);
+		event.target.reset();
+	};
+
+	return (
+		<StyledProfile>
+			<label>
+				<Button greyButton={true} onClick={changeActiveUser}>
+					Switch user
+				</Button>
+			</label>
+			<article id="userInformation">
+				{toggle ? (
+					<h1>{'Username: ' + activeUser.userName}</h1>
+				) : (
+					<Form
+						onSubmit={event => {
+							setNewUsername(event);
+							setToggle(!toggle);
+							setOpen(!open);
+							setTimeout(() => {
+								setOpen(false);
+							}, 1000);
+						}}
+						placeholderText={activeUser.userName}
+						submitButtonText={'Change your Username'}
+					/>
+				)}
+				<Dialog open={open}>
+					<h3>Username changed</h3>
+				</Dialog>
+				<Button
+					id="bigEditButton"
+					editButtonBig={true}
+					onClick={() => {
+						setToggle(!toggle);
+					}}
+				>
+					<svg
```

Comment:
Please have a look at this short video about svg's as React components: https://neuefische-students.slack.com/archives/C03BYPS4ESJ/p1657094592943949

It would make your code a bit less bloated :)

Reply:
I´ve seen the video already and its great! But unfortunately i have very complex svgs (with animation).
I am already talking to dominik to figure out how to build these as component :)